examples/hosts/app-integration/external-data/src/view/taskListView.tsx

```diff
@@ -2,66 +2,64 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
-
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
 import { CollaborativeInput } from "@fluid-experimental/react-inputs";
 
 import React, { useEffect, useRef, useState } from "react";
 
-import type { ITask, ITaskList } from "../model-interface";
+import type { ExternalSnapshotTask, ITask, ITaskList } from "../model-interface";
 
 interface ITaskRowProps {
 	readonly task: ITask;
-	readonly deleteTask: () => void;
+	readonly deleteDraftTask: () => void;
 }
 
 /**
  * The view for a single task in the TaskListView, as a table row.
  */
 const TaskRow: React.FC<ITaskRowProps> = (props: ITaskRowProps) => {
-	const { task, deleteTask } = props;
+	const { task, deleteDraftTask } = props;
 	const priorityRef = useRef<HTMLInputElement>(null);
-	const [incomingName, setIncomingName] = useState<string | undefined>(task.incomingName);
-	const [incomingPriority, setIncomingPriority] = useState<number | undefined>(
-		task.incomingPriority,
+	const [externalDataSnapshot, setExternalDataSnapshot] = useState<ExternalSnapshotTask>(
+		task.externalDataSnapshot,
 	);
-	const [incomingType, setIncomingType] = useState<string | undefined>(task.incomingType);
+	const [showConflictUI, setShowConflictUI] = useState<boolean>(false);
 	useEffect(() => {
 		const updateFromRemotePriority = (): void => {
 			if (priorityRef.current !== null) {
-				priorityRef.current.value = task.priority.toString();
+				priorityRef.current.value = task.draftPriority.toString();
 			}
 		};
-		const showIncomingPriority = (): void => {
-			setIncomingPriority(task.incomingPriority);
-			setIncomingType(task.incomingType);
-		};
-		const showIncomingName = (): void => {
-			setIncomingName(task.incomingName);
-			setIncomingType(task.incomingType);
+		const updateExternalSnapshotData = (value: boolean): void => {
+			setExternalDataSnapshot(task.externalDataSnapshot);
+			setShowConflictUI(value);
 		};
-		task.on("priorityChanged", updateFromRemotePriority);
-		task.on("incomingPriorityChanged", showIncomingPriority);
-		task.on("incomingNameChanged", showIncomingName);
+		task.on("draftPriorityChanged", updateFromRemotePriority);
+		task.on("changesAvailable", updateExternalSnapshotData);
```

Comment:
Why do we need the priority change as a separate flow?

Reply:
The draftPriorityChanged here is completley separate from the external data logic flow. This event is triggered when the little up and down arrows are clicked on the UI and this ecent listening flow renders the numbers on the screen in response to up arrow being clicked or down arrow to increase or decrease the priority.

I agree that the name is confusing and makes it sound like it's related to our external data stuff. But I haven't found a good name to refactor to yet.

Reply:
Ah, it's a local update to the (draft) priority, I see. In that case we probably want to rename the listener to not have "remote".